File: src/main/reporting-app-react/src/services/reportService.js
import axios from "axios";

export default class ReportService {

    getReportTotalPages(size) {
        return axios.get(`http://localhost:8080/report/totalPages?pageSize=${size}`, size)
    }

    doesReportExist(fileNumber) {
        return axios.get(`http://localhost:8080/report/exists?fileNumber=${fileNumber}`, fileNumber)
    }

    getReports(page, size) {
        return axios.get(`http://localhost:8080/report/getAllActiveReport?page=${page}&size=${size}`, page, size)
    }

    getByAscendingReport(page, size) {
        return axios.get(`http://localhost:8080/report/getByAscendingReport?page=${page}&size=${size}`, page, size)
    }

    getByDescendingReport(page, size) {
        return axios.get(`http://localhost:8080/report/getByDescendingReport?page=${page}&size=${size}`, page, size)
    }

    getByReportId(id) {
        return axios.get(`http://localhost:8080/report/getByReportId?reportId=${id}`);
    }

    addReport(report) {
        return axios.post("http://localhost:8080/report/add", report)
    }

    updateReport(report) {
        return axios.put("http://localhost:8080/report/update", report)
    }

    deleteReport(id) {
        return axios.delete(`http://localhost:8080/report/delete?id=${id}`)
    }

    getByPatientIdNumber(patientIdNumber, page, size, asc) {
        return axios.get(`http://localhost:8080/report/getByPatientIdNumber?patientIdNumber=${patientIdNumber}&page=${page}&size=${size}&asc=${asc}`);
    }

    getByPatientName(patientFirstName, patientLastName, page, size, asc) {
        return axios.get(`http://localhost:8080/report/getByPatientName?patientFirstName=${patientFirstName}&patientLastName=${patientLastName}&page=${page}&size=${size}&asc=${asc}`);
    }

    getByLaborantName(laborantFirstName, laborantLastName, page, size, asc) {
        return axios.get(`http://localhost:8080/report/getByLaborantName?laborantFirstName=${laborantFirstName}&laborantLastName=${laborantLastName}&page=${page}&size=${size}&asc=${asc}`);
    }
}

File: src/main/java/GorkemKoc/reportingApplication/business/abstracts/ReportService.java
package GorkemKoc.reportingApplication.business.abstracts;

import GorkemKoc.reportingApplication.business.reponses.*;
import GorkemKoc.reportingApplication.business.requests.CreateReportRequest;
import GorkemKoc.reportingApplication.business.requests.UpdateReportRequest;
import org.springframework.data.domain.Page;
public interface ReportService {
    int getTotalPages(int pageSize);
    boolean doesReportExist(String fileNumber);
    Page<GetAllReportResponse> getAll(int page, int size);
    Page<GetAllActiveReportResponse> getAllActiveReports(int page, int size);
    GetByReportIdResponse getByReportId(int reportId);
    Page<GetByPatientIdNumberResponse>  getByPatientIdNumber(String patientIdNumber, int page, int size, Boolean asc);
    Page<GetByAscendingReportResponse>  getByAscendingReport(int page, int size);
    Page<GetByDescendingReportResponse>  getByDescendingReport(int page, int size);
    Page<GetByPatientNameResponse> getByPatientName(String patientFirstName, String patientLastName, int page, int size, Boolean asc);
    Page<GetByLaborantNameResponse> getByLaborantName(String laborantFirstName, String laborantLastName, int page, int size, Boolean asc);
    void add(CreateReportRequest createReportRequest);
    void update(UpdateReportRequest updateReportRequest);
    void delete(int id);

}
